tools/autograd/derivatives.yaml

```diff
@@ -306,7 +306,7 @@
   self: not_implemented("inplace version of asinh")
 
 - name: atanh(Tensor self) -> Tensor
-  self: grad * 1 / (1 - self.pow(2)).conj()
+  self: grad / (1 - self * self).conj()
```

Comment:
The `* 1` sounds fine.

The conversion from pow to mul would need benchmarking.
In particular we need to make sure that we're not doubling the memory loads here.

Reply:
Thanks @albanD , is there a CI or some way to collect this data ? Also what command to run for benchmarking binary operators on cpu and cuda

Reply:
For benchmarking you can use the `Timer` class from  https://pytorch.org/docs/stable/benchmark_utils.html, and here's a more in depth tutorial https://pytorch.org/tutorials/recipes/recipes/benchmark.html. For CUDA don't forget to call `torch.cuda.synchronize`

Reply:
To check memory usage you can use https://pytorch.org/docs/stable/generated/torch.cuda.memory_allocated.html#torch.cuda.memory_allocated

Reply:
Also do you run into any issues with complex numbers @kulinseth, they seemed to have run into some issues in https://github.com/pytorch/pytorch/pull/84569

(the other issue mentioned in that PR was about in-place, but we're doing out of place here, so that should be fine)